src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardUsersPage.java

```diff
@@ -0,0 +1,187 @@
+        if (userListResult==null){
+            try {
+                throw new Exception("userListResult should not be null!");
+            } catch (Exception ex) {
+                Logger.getLogger(DashboardUsersPage.class.getName()).log(Level.SEVERE, null, ex);
+            }
+        }
+        setSelectedPage(userListResult.getSelectedPageNumber());        
+
+        this.userList = userListResult.getUserList();
+        this.pager = userListResult.getPager();
+        
+        return true;
+        
+    }
+
+
+    
+    public String getListUsersAPIPath() {
+        //return "ok";
+        return Admin.listUsersFullAPIPath;
+    }
+
+    /** 
+     * Number of total users
+     * @return 
+     */
+    public String getUserCount() {
+
+        return NumberFormat.getInstance().format(userService.getTotalUserCount());
+    }
+
+    /** 
+     * Number of total Superusers
+     * @return 
+     */
+    public Long getSuperUserCount() {
+        
+        return userService.getSuperUserCount();
+    }
+
+    public List<AuthenticatedUser> getUserList() {
+        return this.userList;
+    }
+
+    /**
+     * Pager for when user list exceeds the number of display rows
+     * (default: UserListMaker.ITEMS_PER_PAGE)
+     * 
+     * @return 
+     */
+    public Pager getPager() {
+        return this.pager;
+    }
+
+    public void setSelectedPage(Integer pgNum){
+        if ((pgNum == null)||(pgNum < 1)){
+            this.selectedPage = 1;
+        }
+        selectedPage = pgNum;
+    }
+
+    public Integer getSelectedPage(){
+        if ((selectedPage == null)||(selectedPage < 1)){
+            setSelectedPage(null);            
+        }
+        return selectedPage;
+    }
+    
+    public String getSearchTerm() {
+        return searchTerm;
+    }
+
+    public void setSearchTerm(String searchTerm) {
+        this.searchTerm = searchTerm;
+    }
+    
+    /* 
+       Methods for toggling the supeuser status of a selected user. 
+       Our normal two step approach is used: first showing the "are you sure?" 
+       popup, then finalizing the toggled value. 
+    */
+       
+    AuthenticatedUser selectedUser = null; 
+    
+    public void setSelectedUser(AuthenticatedUser user) {
+        this.selectedUser = user;
+    }
+    
+    public AuthenticatedUser getSelectedUser() {
+        return this.selectedUser;
+    }
+    
+    
+    public void setUserToToggleSuperuserStatus(AuthenticatedUser user) {
+        logger.info("selecting user "+user.getIdentifier());
+        selectedUser = user; 
+    }
+    public void saveSuperuserStatus(){
+        logger.info("Toggling user's "+selectedUser.getIdentifier()+" superuser status; (current status: "+selectedUser.isSuperuser()+")");
+        logger.info("Attempting to save user "+selectedUser.getIdentifier());
+        authenticationService.update(selectedUser);
```

Comment:
As of 7a25089 I'm getting this error when I try to make my "spruce" user, who was created via API, a superuser:

`Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "createdtime" violates not-null constraint  Detail: Failing row contains (5, null, null, spruce@mailinator.com, null, Sabrina, null, null, Spruce, null, t, spruce).`

@bsilverstein95 @raprasad @scolapasta I'm assuming this is somehow related to pull request #3931.